src/stores/useHolochainStore.js

```diff
@@ -104,21 +106,107 @@ const makeUseHolochainStore = ({ installed_app_id, app_ws_url, is_hpos_served, h
 
       return result
     },
+    async zomeCall(args) {
+      const zomeCallArgs = {
+        appId: installed_app_id,
+        roleId: args.role_name,
+        zomeName: args.zome_name,
+        fnName: args.fn_name,
+        payload: args.payload
+      }
+
+      const response = await this.hposHolochainCall({path: 'zome_call', headers: {}, params: zomeCallArgs})
+      return response
+    },    
     setCredentials(cellId) {
       this.signingCredentials = new Promise(async (resolve, reject) => {
-        try {
-          const adminWs = await AdminWebsocket.connect(`ws:localhost:${hc_admin_port}`)
-          await adminWs.authorizeSigningCredentials(cellId)
-        } catch(e) {
-          console.log(`holochainCallZome error authorizeSigningCredentials AdminWebsocket: ws:localhost:${hc_admin_port}`, e)
-          reject()
+        if( !is_hpos_served ) { // If running a raw holochain we need to authorize zome calls once
+          try {
+            const adminWs = await AdminWebsocket.connect(`ws:localhost:${hc_admin_port}`)
+            await adminWs.authorizeSigningCredentials(cellId)
+          } catch(e) {
+            console.log(`holochainCallZome error authorizeSigningCredentials AdminWebsocket: ws:localhost:${hc_admin_port}`, e)
+            reject()
+          }
+
+          resolve()
+        } else {
+          try {
+            const [keyPair, signingKey] = await generateSigningKeyPair()
+            const params = { cellId, signingKey }    
+            const cap_token = await this.hposHolochainCall({path: 'cap_token', headers: {}, params})
+
+            const signingCredentials = {
+              capSecret: new Uint8Array(listify(cap_token, (_, value) => (Number(value)))),
+              keyPair,
+              signingKey
+            }
+
+            await setSigningCredentials(cellId, signingCredentials)
```

Comment:
I'm surprised by this. Don't we get zome call signing for free when using hpos holochain api?

https://github.com/Holo-Host/holo-nixpkgs/blob/69f93c029c571753dd5d87b79f8194cd691550ff/overlays/holo-nixpkgs/hpos-holochain-api/src/index.js#L809C1-L811C46

Reply:
You're correct, I removed it.